internal/codeintel/stores/dbstore/repos.go

```diff
@@ -31,3 +32,61 @@ const repoNameQuery = `
 -- source: enterprise/internal/codeintel/stores/dbstore/repos.go:RepoName
 SELECT name FROM repo WHERE id = %s
 `
+
+type GetJVMDependencyReposOpts struct {
+	ArtifactName string
+	After        int
+	Limit        int
+}
+
+type JVMDependencyRepo struct {
+	Module  string
+	Version string
+	ID      int
+}
+
+func (s *Store) GetJVMDependencyRepos(ctx context.Context, filter GetJVMDependencyReposOpts) ([]JVMDependencyRepo, error) {
+	conds := []*sqlf.Query{sqlf.Sprintf("scheme = %s", "semanticdb")}
+
+	var limit interface{} = "ALL"
+
+	if filter.ArtifactName != "" {
+		conds = append(conds, sqlf.Sprintf("name = %s", filter.ArtifactName))
+	}
+
+	if filter.After > 0 {
+		conds = append(conds, sqlf.Sprintf("id > %d", filter.After))
+	}
+
+	if filter.Limit != 0 {
+		limit = filter.Limit
+	}
+
+	return scanJVMDependencyRepo(s.Query(ctx, sqlf.Sprintf(getLSIFDependencyReposQuery, sqlf.Join(conds, "AND"), limit)))
+}
+
+func scanJVMDependencyRepo(rows *sql.Rows, queryErr error) (dependencies []JVMDependencyRepo, err error) {
+	if queryErr != nil {
+		return nil, queryErr
+	}
+	defer func() { err = basestore.CloseRows(rows, err) }()
+
+	for rows.Next() {
+		var dep JVMDependencyRepo
+		if err = rows.Scan(
+			&dep.Module,
+			&dep.Version,
+		); err != nil {
+			return nil, err
+		}
+
+		dependencies = append(dependencies, dep)
+	}
+
+	return dependencies, nil
+}
+
+const getLSIFDependencyReposQuery = `
+SELECT id, name, version FROM lsif_dependency_repos
+WHERE %s ORDER BY id LIMIT %v
```

Comment:
```suggestion
-- source: internal/codeintel/stores/dbstore/repos.goo:GetJVMDependencyRepos
SELECT id, name, version FROM lsif_dependency_repos
WHERE %s ORDER BY id %s
```

Reply:
Should it be the following instead? I made the query generic for other dependency types in the near future 
```suggestion
-- source: internal/codeintel/stores/dbstore/repos.go:GetLSIFDependencyRepos
SELECT id, name, version FROM lsif_dependency_repos
WHERE %s ORDER BY id %s
```

Reply:
Oops did I quote the wrong caller? I'd like to have some source (even if it's an over-approximation). Could we at least put the filename, or extend it with something like `repos.go:{caller1,caller2,caller3}` in the future?

Reply:
The caller wasnt wrong per se, its that the query is generic to allow also GetNPMDependencyRepos to use it, so there would be multiple callers of that query. Im cool with both of the last sentence :+1: